realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java

```diff
@@ -115,6 +116,86 @@ public void create_duplicatedNameThrows() {
     }
 
     @Test
+    public void createWithPrimaryKeyField_string() {
+        // Not nullable
+        realmSchema.createWithPrimaryKeyField("FooNonNull", "pkField", String.class, FieldAttribute.REQUIRED);
+        RealmObjectSchema objectSchema = realmSchema.getSchemaForClass("FooNonNull");
```

Comment:
we didn't overwrite the `RealmObjectSchema.equals()`, so the `assertSame` will have no point here i think?